src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -715,4 +715,45 @@
       it { is_expected.to redirect_to(:back) }
     end
   end
+
+  describe 'before_filter set_maintained_project' do
+    before do
+      login user
+      user.home_project.update(kind: 'maintenance')
+    end
+
+    it "#remove_maintained_project raise excepction with invalid maintained project" do
+      expected = expect do
+        post :remove_maintained_project, project: user.home_project, maintained_project: "invalid"
+      end
+      expected.to raise_exception ActiveRecord::RecordNotFound
+    end
+  end
+
+  describe "before_filter require_maintenance_project" do
+    let(:maintained_project) { create(:maintained_project, project: user.home_project) }
+
+    before do
+      login user
+    end
+
+    context "#remove_maintained_project fails without maintenance kind for a valid maintained project" do
+      before do
+        request.env["HTTP_REFERER"] = root_url # Needed for the redirect_to :back
+        post :remove_maintained_project, project: user.home_project, maintained_project: maintained_project.project.name
+      end
+
+      it { is_expected.to redirect_to(:back) }
+    end
+
+    context "#remove_maintained_project succeeds with maintenance kind for a valid maintained project" do
+      before do
+        user.home_project.update(kind: 'maintenance')
+        post :remove_maintained_project, project: user.home_project, maintained_project: maintained_project.project.name
+      end
+
+      it { expect(flash[:notice]).to eq("Removed #{maintained_project.project.name} from maintenance") }
+      it { is_expected.to redirect_to(action: 'maintained_projects', project: user.home_project) }
+    end
+  end
```

Comment:
@Ana06 @ChrisBr I don't think we should have a separate controller test for before filters. Maybe in some rare edge cases like authentification related before filters. But not here.

This seems to be implementation details of how we handle the corresponding http request. So I think it would be better to integrate these tests into the other "POST #remove_maintained_project" part.
And if it turns out that we do not catch the ActiveRecord::RecordNotFound and show a proper error message, we should fix this:-)